sklearn/tree/tree.py

```diff
@@ -762,12 +764,18 @@ class DecisionTreeClassifier(ClassifierMixin, BaseDecisionTree):
     reduce memory consumption, the complexity and size of the trees should be
     controlled by setting those parameter values.
 
-    The features are always randomly permuted at each split. Therefore,
-    the best found split may vary, even with the same training data and
-    ``max_features=n_features``, if the improvement of the criterion is
-    identical for several splits enumerated during the search of the best
-    split. To obtain a deterministic behaviour during fitting,
-    ``random_state`` has to be fixed.
+    The features are always randomly permuted at each split, even if 
+    ``splitter`` is set to ``"best"``. Especially if 
+    ``max_features < n_features``, the algorithm will select ``max_features`` 
+    at random at each split. But the best found split may vary accross different 
+    runs, even if ``max_features=n_features``. That is the case, if the 
+    improvement of the criterion is identical for several splits and one split
+    has to be selected at random. 
+    To obtain a deterministic behaviour during fitting, ``random_state`` has to 
+    be fixed. To fix ``random_state`` it has to be set to an arbitrary integer.  
```

Comment:
```suggestion
    be fixed to an int.
```

If you follow my suggestion from above maybe we can remove the rest which is redundant with the glossary.

Reply:
Thanks for the review @NicolasHug. #15222 makes a lot of sense. I integrated a link to the glossary and I integrated your comments and the comments by adrinjalali into the PR.

Reply:
Sorry, but do I have to merge the current master into my branch in order to address the failing ci checks?

Reply:
yes, you can do that by running `git pull upstream master` in you PR branch.

Reply:
OK, thanks. I did that...